monai/utils/deprecated.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2020 - 2021 MONAI Consortium
```

Comment:
looks nice, does this global lock work well with the multiprocess loader? I'll double check soon. would be great to automatically update `object.__doc__` property for the message, but it could be done in the follow up PRs.